listening-comp/backend/db_operations.py

```
@@ -0,0 +1,575 @@
+        """Restore database from a backup."""
+        # Make sure the backup exists
+        if not os.path.exists(backup_path):
+            raise FileNotFoundError(f"Backup not found at {backup_path}")
+        
+        # Create a backup of the current database first
+        self.backup_database()
+        
+        # Close any open connections
+        # This is best-effort; in a real app you'd want connection pooling
+        try:
+            with self.get_connection() as conn:
```

Ruff (0.8.2)

Local variable `conn` is assigned to but never used

Remove assignment to unused variable `conn`

(F841)